jsapp/js/components/modalForms/encryptForm.es6

```diff
@@ -0,0 +1,157 @@
+import autoBind from 'react-autobind';
+import React from 'react';
+import reactMixin from 'react-mixin';
+import Reflux from 'reflux';
+import TextBox from 'js/components/textBox';
+
+import {actions} from 'js/actions';
+import {bem} from 'js/bem';
+import {MODAL_TYPES} from 'js/constants';
+import {stores} from 'js/stores';
+import {t} from 'utils';
+
+class EncryptForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    let submissionURL, publicKey;
+    if (props.asset) {
+      submissionURL = props.asset.content.settings.submission_url;
+      publicKey = props.asset.content.settings.public_key;
+    }
+
+    this.state = {
+      asset: props.asset,
+      assetUid: props.assetUid,
+      submissionURL: submissionURL || '',
+      publicKey: publicKey || '',
+      clearEncryption: false,
+    };
+
+    autoBind(this);
+  }
+
+  componentDidMount() {
+    this.listenTo(stores.asset, this.onAssetsChange);
+
+    if (!this.state.asset && this.state.assetUid) {
+      if (stores.asset.data[this.state.assetUid]) {
+        this.onAssetChange(stores.asset.data[this.state.assetUid]);
+      } else {
+        stores.allAssets.whenLoaded(this.props.assetUid, this.onAssetChange);
+      }
+    }
+  }
+
+  onAssetChange(asset) {
+    this.setState({
+      asset: asset,
+      submissionURL: asset.content.settings.submission_url,
+      publicKey: asset.content.settings.public_key
+    });
```

Comment:
When asset loads we can assume it was because it finished setting the encryption, so here you can set `isPending` to `false` :)